javascript-cont/vscode-and-style.md

```diff
@@ -71,139 +71,260 @@ Hello Naya Spence!
 - [Google](https://google.github.io/styleguide/jsguide.html) also has a popular coding style guide.
 
-We will use the linting setting [ESLint Standard](https://eslint.org/docs/rules/) and [ESLint-react](https://github.com/yannickcr/eslint-plugin-react) which have a rather minimal set of linting rules to highlight common JavaScript problems and will continue to apply when we begin working with React.
+We will use the linting setting [ESLint Standard](https://eslint.org/docs/rules/) and [ESLint-React](https://github.com/yannickcr/eslint-plugin-react) (after introducing the React framework), which have a fairly minimal set of linting rules to highlight common JavaScript problems.
+
+### Applying Style Rules to Our Code
+
+Eslint reports violations of a style guide. We can then fix our code to follow our style rules. We encourage actively watching for Eslint errors, which appear both as wavy underlines and in the Problems VS Code tab, and fixing them as we write code. This trains our eyes to see style issue before they arise, while helping us develop a clean, consistent personal coding style.
+
+On teams, however, this manual approach can lead to problems. It's possible to overlook errors, allowing style guide violations to be checked in to our codebase. To help with this, code formatters exist that can automatically reformat our code any time we save. When starting out, this can be jarring and frustrating. But it's often worth it when working with others to ensure that everyone really is working according to the style guide.
+
+On our own, it's not as necessary to go completely automated. We can still use a code formatter, but we can also choose to run it manually when we want to. This can be a good way to learn about the style guide and how it works, and to see how our code changes as we apply the formatter.
+
+<!-- available callout types: info, success, warning, danger, secondary, star  -->
+
+### !callout-danger
 
-## Typical Node Project Structure
+## Use the Same Style Guide as Your Team!
 
-A typical Node.js project will have the following structure:
+If automatically applying style guides when saving, it's _essential_ that everyone work from the same style guide. Otherwise, the conflicting style guides will cause the formatter to repeatedly reformat code that was written according to a different style guide, and we'll be back to the same problem of unnecessary churn and conflict.
+
+<br/>
+
+Changes caused by a formatter can make it difficult to track changes in version control, as the important part of a commit can be hard to distinguish from the noise introduced by the formatter. If we notice that previously committed code isn't following the style guide, it's better to put any logic changes in one commit, and the style fixes in a separate commit that is clearly labeled as a styling pass commit. This makes it easier to see what's going on in the commit history.
```

Comment:
It would be nice if the sentence were updated to read like this for clarity: "If we notice that previously committed code isn't following the style guide, it's better to put any logic changes in one commit, and the style fixes in a separate commit that is clearly labeled to indicate that the changes were made to comply with styling rules."

Maybe nitpicky and I'd be ok with the way the sentence is written but "styling pass commit" reads a little like industry buzz words and didn't want it to confuse students. Up to you tho!